### ISSUE TITLE
Can you help resolve this

### ISSUE THREAD

**Issue by yassinevic** [skotz/cbl-js#41]
Its a five number captcha. this is where I get so far.
        var cbl = new CBL({
            preprocess: function(img) {
				img.binarize(32);
                img.cropRelative(23, 2, 50, 2);
				img.colorRegions(50, true, 1);
            },
            /* Load the model we saved during training. */
            model_file: "model.txt",
            character_set: "0123456789abcdefghijklmnopqrstuvwxyz",
			exact_characters_width: 2,
            exact_characters: 5,
            blob_min_pixels: 1,
            blob_max_pixels: 100000,
            pattern_width: 30,
            pattern_height: 24,
            allow_console_log: true,
            blob_console_debug: true,
            /* Define a method that fires immediately after successfully loading a saved model. */
            model_loaded: function() {
                // Don't enable the solve button until the model is loaded.
                document.getElementById('solve').style.display = "block";
            }
        });    
        
        var solve = function() {
            // Using the saved model, attempt to find a solution to a specific image.
            cbl.solve("captcha").done(function (solution) {
                // Upon finding a solution, fill the solution textbox with the answer.
                document.getElementById('solution').value = solution;
            });
        }

![0](https://user-images.githubusercontent.com/3239581/68757408-3104e500-060c-11ea-9e5e-8c10c8b6a4db.png)
![1](https://user-images.githubusercontent.com/3239581/68757409-3104e500-060c-11ea-829b-3f46b34ca6e9.png)
![2](https://user-images.githubusercontent.com/3239581/68757410-3104e500-060c-11ea-9cb5-5a23454d10f6.png)
![3](https://user-images.githubusercontent.com/3239581/68757411-319d7b80-060c-11ea-8377-2d385051fafb.png)
![4](https://user-images.githubusercontent.com/3239581/68757412-319d7b80-060c-11ea-8391-f2949df88f45.png)
![5](https://user-images.githubusercontent.com/3239581/68757413-32361200-060c-11ea-98d0-62776f04877e.png)
![6](https://user-images.githubusercontent.com/3239581/68757414-32361200-060c-11ea-8b81-da5708dbb1b2.png)
![7](https://user-images.githubusercontent.com/3239581/68757415-32cea880-060c-11ea-852d-fa561c4e22c4.png)
![8](https://user-images.githubusercontent.com/3239581/68757416-32cea880-060c-11ea-8471-1e7cae92a9e0.png)
![9](https://user-images.githubusercontent.com/3239581/68757418-32cea880-060c-11ea-85a6-667ea7c1288e.png)

Tnx a lot

**Comment by skotz** [comment]
Yeah, lines can be difficult to remove... I'd recommend playing around with convolution filters after the `binarize` step to see if you can filter away the horizontal lines. Maybe start with something like this.

```
img.convolute([ [-1,  4, -1],
                [-1,  2, -1],
                [-1,  4, -1] ], 1);
```

Other than that I'd probably have to extend the library with a method to specifically detect and remove horizontal lines.

**Comment by skotz** [comment]
I've added a new method to remove lines. It's experimental, so it might have some bugs, It's not perfect but it works decently.

![image](https://user-images.githubusercontent.com/11474654/69293252-f2ec6000-0bcd-11ea-92d9-9b1d77bd3107.png)

```
var cbl = new CBL({
    preprocess: function(img) {
        img.debugImage("debugPreprocessed");
        img.cropRelative(20, 2, 20, 2);
        img.debugImage("debugPreprocessed");
        img.removeLight(16);
        img.debugImage("debugPreprocessed");
        img.removeHorizontalLine(2);
        img.debugImage("debugPreprocessed");
        img.colorRegions(50, true, 0);
        img.debugImage("debugPreprocessed");
    },
    character_set: "0123456789abcdefghijklmnopqrstuvwxyz",
    exact_characters_width: 18,
    exact_characters_play: 3,
    exact_characters: 5,
    blob_min_pixels: 50,
    blob_max_pixels: 10000,
    pattern_width: 30,
    pattern_height: 30,
    allow_console_log: true,
    blob_console_debug: true,
    perceptive_colorspace: true,
    blob_debug: "debugSegmented"
});
```
